Sources/TSCBasic/Process.swift

```diff
@@ -677,24 +688,40 @@ public final class Process {
     /// Blocks the calling process until the subprocess finishes execution.
     @discardableResult
     public func waitUntilExit() throws -> ProcessResult {
+        let group = DispatchGroup()
+        group.enter()
+        var processResult : Result<ProcessResult, Swift.Error>?
+        self.waitUntilExit() { result in
+            processResult = result
+            group.leave()
+        }
+        group.wait()
+        switch processResult! {
+        case .failure(let error):
+            throw error
+        case .success(let result):
+            return result
+        }
```

Comment:
can be simplified to `return try processResult.get()`

Reply:
Thanks for the tips. I'm still learning this pattern.